### PR TITLE
docs: Refreshing GlobalHeader Storybook Example

### PR DESCRIPTION
**Pull request by williamjstanton**
## Summary

Fixes: none 

The Storybook example `GlobalHeader` had some opportunities for improved accessibility. 
1. Added accessible `Tooltip` to icon buttons
2. Replaced old `SearchField` with `Combobox`
3. Used `AriaLiveRegion` to demonstrate screen reader supported `CountBadge`
4. Updated style packages

## Release Category
Documentation, Examples 

### Release Note
Optional release note message. Changelog and release summaries will contain a pull request title. This section will add additional notes under that title. This section is not a summary, but something extra to point out in release notes. An example might be calling out breaking changes in a labs component or minor visual changes that need visual regression updates. Remove this section if no additional release notes are required.

## Checklist

- [ ] MDX documentation adheres to Canvas Kit's [Documentation Guidelines](https://workday.github.io/canvas-kit/?path=/docs/guides-documentation-guidelines--page)
- [ ] Label `ready for review` has been added to PR

## For the Reviewer

- [ ] PR title is short and descriptive
- [ ] PR summary describes the change (Fixes/Resolves linked correctly)
- [ ] PR Release Notes describes additional information useful to call out in a release message or removed if not applicable
- [ ] Breaking Changes provides useful information to upgrade to this code or removed if not applicable

## Areas for Feedback? (optional)

- [ ] Code
- [ ] Documentation
- [ ] Testing
- [ ] Codemods

### CODE REVIEW

File: modules/react/_examples/stories/examples/GlobalHeader.tsx [hunk 1]
```diff
@@ -1,41 +1,161 @@
...
 interface HeaderItemProps extends FlexProps {}
+interface LiveCountBadgeProps extends FlexProps {
+  cnt: number;
+}
+
+const tasks = ['Request Time Off', 'Create Expense Report', 'Change Benefits'];
+
+function negate(value: string, fallback?: string) {
+  return `calc(${cssVar(value, fallback)} * -1)`;
+}
```
Comment (mannycarrera4) [inline modules/react/_examples/stories/examples/GlobalHeader.tsx:42]: I think we have a negate on `calc` ?

File: modules/react/_examples/stories/examples/GlobalHeader.tsx [hunk 2]
```diff
@@ -1,41 +1,161 @@
...
+const styleOverrides = {
+  headerWrapper: createStyles({
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    boxSizing: 'border-box',
+    ...type.levels.subtext.large,
+    WebkitFontSmoothing: 'antialiased',
+    MozOsxFontSmoothing: 'grayscale',
+    backgroundColor: colors.frenchVanilla100,
```
Comment (mannycarrera4) [inline modules/react/_examples/stories/examples/GlobalHeader.tsx:53]: Let's update this values to use your new tokens `system.color.bg.default`. Here's a renferece: https://workday.github.io/canvas-tokens/?path=/story/visual-tests-system-tokens-colors--background

File: modules/react/_examples/stories/examples/GlobalHeader.tsx [hunk 3]
```diff
@@ -1,41 +1,161 @@
...
+    ...depth[1],
+    padding: space.xxs,
```
Comment (mannycarrera4) [inline modules/react/_examples/stories/examples/GlobalHeader.tsx:55]: same here: let's use our system tokens for styling

File: modules/react/_examples/stories/examples/GlobalHeader.tsx [hunk 4]
```diff
@@ -1,41 +1,161 @@
...
+    color: colors.blackPepper500,
+  }),
+  notificationContainerStyles: createStyles({
+    boxSizing: 'border-box',
+    position: 'relative',
+  }),
+  countBadgeStyles: createStyles({
+    boxSizing: 'border-box',
+    position: 'absolute',
+    top: negate(system.space.x1),
```
Comment (mannycarrera4) [inline modules/react/_examples/stories/examples/GlobalHeader.tsx:80]: here's where you can call `calc.negate`

File: modules/react/_examples/stories/examples/GlobalHeader.tsx [hunk 5]
```diff
@@ -1,41 +1,161 @@
...
+              MENU
+            </TertiaryButton>
+          </Tooltip>
+          <Tooltip title="Workday Home">
+            <TertiaryButton>
+              <img src="https://design.workday.com/images/ck-dub-logo-blue.svg" alt="" />
+            </TertiaryButton>
+          </Tooltip>
+        </GlobalHeader.Item>
+        <GlobalHeader.Item margin="auto" width="100%" maxWidth={`calc(${space.xxxl} * 6)`}>
```
Comment (mannycarrera4) [inline modules/react/_examples/stories/examples/GlobalHeader.tsx:133]: We've been updating our examples to move away from style props. You can have your `GlobalHeader.Item` extend `CSProps` in the interface, and then use `<GlobalHeader.Item cs={{maxWidth: calc.multiply(space.xxxl, * 6)}>`

File: modules/react/_examples/stories/examples/GlobalHeader.tsx [hunk 6]
```diff
@@ -45,21 +165,89 @@ const GlobalHeaderItem = createComponent('div')({
...
+            value={searchText}
+            {...props}
+          />
+        </InputGroup>
+        <Combobox.Menu.Popper>
+          <Combobox.Menu.Card>
+            {filteredTasks.length === 0 ? (
+              <StyledMenuItem as="span">No Results Found</StyledMenuItem>
+            ) : (
+              filteredTasks.map(i => (
```
Comment (mannycarrera4) [inline modules/react/_examples/stories/examples/GlobalHeader.tsx:208]: Although mapping works, we should probably use a render function like in the auto complete example:

```
{model.state.items.length > 0 && (
                <Combobox.Menu.List cs={styleOverrides.comboboxMenuList}>
                  {item => <Combobox.Menu.Item>{item}</Combobox.Menu.Item>}
                </Combobox.Menu.List>
              )}
```

**Review by mannycarrera4 — CHANGES_REQUESTED** [review]
Lets clean up some of the styles and code examples. Content wise it's great! Thank you as always for working through making our documentation better

File: modules/react/_examples/stories/examples/GlobalHeader.tsx (hunk 6 above)
Comment (williamjstanton) [reply modules/react/_examples/stories/examples/GlobalHeader.tsx:207]: I'll need some help with using this model!

File: modules/react/_examples/stories/examples/GlobalHeader.tsx [hunk 7]
```diff
@@ -1,65 +1,252 @@
...
+      </Flex>
+    </>
+  );
+};
+
 const GlobalHeaderItem = createComponent('div')({
   displayName: 'GlobalHeader.Item',
   Component: ({gap = 's', ...props}: HeaderItemProps, ref) => (
-    <Flex gap={gap} alignItems="center" marginX={space.xs} ref={ref} {...props} />
+    <Flex gap={gap} alignItems="center" marginX={system.space.x3} ref={ref} {...props} />
```
Comment (mannycarrera4) [inline modules/react/_examples/stories/examples/GlobalHeader.tsx:161]: does passing a token here work?

**Review by mannycarrera4 — APPROVED** [review]

**Pull request merged**